lib/geoengineer/gps/finder.rb

```diff
@@ -0,0 +1,99 @@
+# Finder is a class that contains all the logic to find a GPS coordinate
+class GeoEngineer::GPS::Finder
+  class NotFoundError < StandardError; end
+  class NotUniqueError < StandardError; end
+  class BadQueryError < StandardError; end
+  class BadReferenceError < StandardError; end
+
+  NODE_QUERY_SYNTAX = %r{
+    ^
+    (?<project>[a-zA-Z0-9\-_/*]*):         # Match the project name (optional)
+    (?<environment>[a-zA-Z0-9\-_*]*):      # Match the environment (optional)
+    (?<configuration>[a-zA-Z0-9\-_*]*):    # Match the configuration (optional)
+    (?<node_type>[a-zA-Z0-9\-_*]+):         # Match the node_type (required), does not support `*`
```

Comment:
this comment is wrong, the regex does support star it looks like

Reply:
true